<a href="https://colab.research.google.com/github/alessaag/cursoIA_Polygonus/blob/main/Clase_N%C2%B011.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **<font color='#FFD92F'>DISTRIBUCIONES DE PROBABILIDAD: "Una Introducción"</font>** 🌎



En el mundo de la probabilidad y la estadística, las **distribuciones de probabilidad** son herramientas fundamentales para comprender el comportamiento de *variables aleatorias*. Son funciones que asignan a cada posible resultado de un experimento aleatorio, la probabilidad de que dicho resultado ocurra.

## ⭐**Tipos de Variables aleatorias**⭐

### **<font color='#FFAA0E'>Discretas:</font>**
Solo pueden tomar un número **`finito o infinito numerable`** de valores. Algunos ejemplos son:

* El número de cartas en una baraja inglesa.

* El número de hijos en una familia.

* El número de defectos en un producto.

In [1]:
from IPython.display import Image

Image(url='https://w7.pngwing.com/pngs/693/394/png-transparent-solitaire-game-application-playing-card-standard-52-card-deck-card-game-deck-of-cards-game-text-rectangle.png', width = 1200, height = 500)

### **<font color='#FFAA0E'>Continuas:</font>**

Pueden tomar cualquier **`valor dentro de un intervalo.`** Algunos ejemplos son:

* La altura de una persona.
* El peso de una caja.
* La temperatura en un día determinado.

In [2]:
Image(url='https://cdn.statcdn.com/Infographic/images/normal/23732.jpeg', width = 600, height = 800)

### 🌳**<font color='#BCBD22'>Según su naturaleza</font>**🌳

> **`Cualitativas:`** Sus valores son categorías o cualidades, como el color de ojos o la marca de un vehículo.

> **`Cuantitativas:`** Sus valores son numéricos, como la altura o el peso.

🏹🎯 **Practica** 🎯🏹

* `El color de un automóvil (rojo, azul, verde, etc.).`

Cualitativa

* `El número de estudiantes en una clase.`

Cuantitativa



* `La temperatura en una ciudad.`

Cuantitativa

* `La altura de un edificio.`

Cuantitativa

## ⭐**Tipos de Distribuciones de Probabilidad**⭐

Las distribuciones de probabilidad son herramientas esenciales para el análisis de datos y la toma de decisiones en diversos campos. La elección de la distribución adecuada depende de las características de la variable aleatoria que se está estudiando.

### **<font color='#98D9E4'>DISTRIBUCIONES DISCRETAS**


#### **<font color ='#30BCAD'> Binomial:**

In [3]:
Image(url='https://upload.wikimedia.org/wikipedia/commons/thumb/7/75/Binomial_distribution_pmf.svg/488px-Binomial_distribution_pmf.svg.png', width = 500, height = 400)

✅ Se utiliza para modelar la probabilidad de obtener **`"éxitos"`** en una serie de **`"n"`** ensayos independientes, con probabilidad constante de éxito **`"P"`**. Se aplica en:

* `Lanzamiento de monedas.`
* `Pruebas de diagnóstico.`
* `Encuestas de opinión`

#### **<font color ='#30BCAD'>Poisson:**

In [10]:
Image(url='https://upload.wikimedia.org/wikipedia/commons/thumb/1/16/Poisson_pmf.svg/488px-Poisson_pmf.svg.png', width = 500, height = 400)

✅ Describe la probabilidad de que **`ocurra un número determinado de eventos en un intervalo de tiempo o espacio`**, dado que la tasa de ocurrencia es constante. Se utiliza en:

* `Número de llamadas telefónicas en un centro de atención al cliente.`
* `Defectos en un producto.`
* `Accidentes de tráfico en una carretera.`

#### **<font color ='#30BCAD'>Distribución uniforme:**

In [11]:
Image(url='https://sf.ezoiccdn.com/ezoimgfmt/www.probabilidadyestadistica.net/wp-content/uploads/2022/12/distribucion-uniforme-discreta.png?ezimgfmt=rs:334x234/rscb1/ngcb1/notWebP', width = 500, height = 350)


✅ **`Asigna la misma probabilidad a cada valor posible dentro de un intervalo.`** Se aplica en:

* `Lanzamiento de dados.`
* `Selección aleatoria de números.`
* `Simulaciones Monte Carlo.`

#### **<font color ='#30BCAD'>Bernoulli:**

In [6]:
Image(url='https://la.mathworks.com/help/stats/computebernoullidistributionpdfexample_01_es.png', width = 600, height = 500)

✅ Es un caso especial de la distribución binomial con "n" = 1. **`Se utiliza para modelar eventos con solo dos posibles resultados, como éxito o fracaso.`** Se aplica en:

* `Funcionamiento de un dispositivo electrónico.`
* `Superación de un examen.`
* `Respuesta a una pregunta de sí/no.`

### **<font color='#98D9E4'>DISTRIBUCIONES CONTINUAS**

#### **<font color ='#30BCAD'>Normal:**

In [7]:
Image(url='https://fhernanb.github.io/Manual-de-R/Manual_de_R_files/figure-html/casesDistNormal-1.png', width = 972, height = 729)

✅ **`También conocida como distribución de Gauss`**, es una de las distribuciones más importantes en estadística. Se caracteriza por su forma de campana simétrica. Se aplica en:

* `Altura de personas.`
* `Peso de animales.`
* `Distribución de calificaciones en un examen.`


#### **<font color ='#30BCAD'>Uniforme:**



In [12]:
Image(url='https://r-coder.com/images/featured/uniform-distribution-in-r.png', width = 800, height = 500)

✅ Similar a la distribución uniforme discreta, pero para variables continuas. Se aplica en:

* `Distribución de la temperatura en un día.`
* `Tiempos de espera en una fila.`
* `Errores de medición.`

#### **<font color ='#30BCAD'>Exponencial:**

In [9]:
Image(url='https://upload.wikimedia.org/wikipedia/commons/thumb/e/ec/Exponential_pdf.svg/450px-Exponential_pdf.svg.png', width = 500, height = 400)

✅ **`Describe el tiempo hasta que ocurre un evento`**, con una tasa de ocurrencia constante. Se utiliza en:

* `Tiempo de vida útil de un componente electrónico.`
* `Tiempo de espera entre llamadas telefónicas.`
* `Tiempo de llegada de clientes a una tienda.`

## ⭐**Aplicaciones en Inteligencia Artificial**⭐

### 📞 **`Contexto:` Un centro de llamadas recibe un promedio de 10 llamadas por hora.**

✅ **`Objetivo:`** Calcular la probabilidad de que el centro de llamadas reciba exactamente 15 llamadas en la próxima hora.

In [22]:
from scipy.stats import poisson

# Parámetros
lambda_ = 10  # Promedio de llamadas por hora
k = 15  # Número de llamadas a calcular

# Calcular la probabilidad
probabilidad = poisson.pmf(k, lambda_)

# Imprimir el resultado
print(f"Probabilidad: {probabilidad:.4f}")

Probabilidad: 0.0347


💡**`Interpretación:`** La probabilidad de que el centro de llamadas reciba exactamente 15 llamadas en la próxima hora es de aproximadamente 3.47%.

💡**`Explicación:`**

La distribución de Poisson se utiliza para modelar la probabilidad de que ocurra un número específico de eventos en un intervalo de tiempo fijo, suponiendo que los eventos ocurren de forma independiente y con una tasa constante.

En este caso, la función ```poisson.pmf``` de la biblioteca ```scipy.stats``` calcula la probabilidad de que ocurran exactamente ```k``` eventos (llamadas) en un intervalo de tiempo (una hora), con una tasa promedio de ```lambda_``` eventos por unidad de tiempo (10 llamadas por hora).



### 👨👩 **`Contexto:` La altura de la población humana tiende a seguir una distribución normal.**



✅**`Objetivo:`** Calcular la probabilidad de que una persona seleccionada al azar mida más de 1.80 metros, suponiendo que la altura promedio es de 1.70 metros y la desviación estándar es de 0.10 metros.

In [23]:
from scipy.stats import norm

# Parámetros
media = 1.70  # Altura promedio
desviacion_estandar = 0.10  # Desviación estándar
altura = 1.80  # Altura a calcular

# Calcular la probabilidad
probabilidad = norm.cdf(altura, media, desviacion_estandar)

# Imprimir el resultado
print(f"Probabilidad: {probabilidad:.4f}")


Probabilidad: 0.8413


💡**`Interpretación:`**

La probabilidad de que una persona seleccionada al azar mida más de 1.80 metros es de aproximadamente 84.13%.

💡**`Explicación:`**

En este caso, la función ```norm.cdf``` de la biblioteca ```scipy.stats``` calcula la probabilidad de que una variable aleatoria con distribución normal sea menor o igual a un valor específico (```altura```). La función toma como argumentos la media (```media```), la desviación estándar (```desviacion_estandar```) y el valor a calcular (```altura```).

## ⛳**Ejercicios de práctica**

📞**Modificaciones:**

Puedes modificar el código para calcular la probabilidad de otros valores de ```k``` (número de llamadas) o para diferentes valores de ```lambda_``` (promedio de llamadas por hora).

In [1]:
from scipy.stats import poisson

# Parámetros
lambda_ = 10  # Promedio de llamadas por hora
k = 8  # Número de llamadas a calcular

# Calcular la probabilidad
probabilidad = poisson.pmf(k, lambda_)

# Imprimir el resultado
print(f"Probabilidad: {probabilidad:.4f}")

Probabilidad: 0.1126


👨👩 **Modificaciones:**

Puedes modificar el código para calcular la probabilidad de que una persona mida entre dos valores específicos, o para diferentes valores de la media y la desviación estándar.

In [2]:
from scipy.stats import norm

# Parámetros
media = 1.70  # Altura promedio
desviacion_estandar = 0.10  # Desviación estándar
altura1 = 1.75  # Altura a calcular
altura2 = 1.64  # Altura a calcular


# Calcular la probabilidad de que la altura esté entre altura2 y altura1
probabilidad = norm.cdf(altura1, media, desviacion_estandar) - norm.cdf(altura2, media, desviacion_estandar)

# Imprimir el resultado
print(f"Probabilidad: {probabilidad:.4f}")

Probabilidad: 0.4172


## 📚 **Referencias Bibliográficas**

> **`Distribuciones de probabilidad`**

* "Introducción a la probabilidad y estadística" por Jay L. Devore.

* "Fundamentos de estadística" por Ronald E. Walpole, Raymond H. Myers, Sharon L. Myers y Keying Ye.

* "Probabilidad y estadística para ingeniería y ciencias" por William W. Hines, Douglas C. Montgomery y David M. Goldsman.

> **`Python para la ciencia de datos:`**

* "Python para la ciencia de datos" por Jake VanderPlas.

* "Hands-On Machine Learning with Scikit-Learn and TensorFlow" por Aurélien Géron.

* "Introducción al análisis de datos con Python" por Wes McKinney.